### ISSUE TITLE
[ENH] - Include "--attempt-fixes" flag from Nebari upgrade CLI in upgrade steps logic

### ISSUE THREAD

**Issue by viniciusdc** [nebari-dev/nebari#2761]
### Feature description

We are currently exposing an `--attempt-fixes` flag in the `nebari upgrade` subcommand: https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/subcommands/upgrade.py#L19-L23

This parameter can be accessed from the `UpgradeStep` class, where we define specific upgrade logic for each upgrade step between contiguous versions: https://github.com/nebari-dev/nebari/blob/9b1310b33e89c2c11c3b39128ec792ca80342486/src/_nebari/upgrade.py#L52-L111

However, we're not using this flag in the most recent upgrade steps. We need to ensure that upgrade steps use this flag and define a logical path to upgrade without user input.

### Value and/or benefit

Recently, https://github.com/nebari-dev/nebari/pull/2780 got merged, implementing the logic to add automatic upgrade tests to our GHA workflows. Because tests are running automatically, we cannot rely on user input, so they are relying on  passing the `--attempt-fixes` flag when calling the `nebari upgrade` command: https://github.com/nebari-dev/nebari/blob/364c9e3d5b37286ab32f9b13542a1d3e2cd6fb0a/.github/workflows/test_local_upgrade.yaml#L73

**Comment by Suraj-kumar00** [comment]
Hi @viniciusdc, how have you been?
As I'm working on DevOps practices, I would love to work on this issue but can you please give me some more context?
What I have understood from you description is that:
- The `--force` flag will automatically accept upgrade prompts, bypassing manual confirmation.
- It is useful for CI pipelines to automate the upgrade process without user intervention.
- It speeds up testing and ensures upgrades can proceed in automated environments.
- Overall, it streamlines CI/CD processes by allowing the upgrade command to run without manual input.

**Comment by marcelovilla** [comment]
Hey @Suraj-kumar00!

Sorry for the late reply and your interest in this issue is very welcome.

> * The --force flag will automatically accept upgrade prompts, bypassing manual confirmation.
> * It is useful for CI pipelines to automate the upgrade process without user intervention.
> * It speeds up testing and ensures upgrades can proceed in automated environments.
> * Overall, it streamlines CI/CD processes by allowing the upgrade command to run without manual input.

Your understanding is correct. If you take a look at the [upgrade.py file](https://github.com/nebari-dev/nebari/blob/215bbd622f0ea1692e79bb4411db81ef28b1d329/src/_nebari/upgrade.py), you'll notice that we have an upgrade step for each version, where we prompt the user depending on what needs to be done. While there are some cases where no action is required from the user, there are other cases where the user needs to decide whether they should nebari to run operations on their behalf. See, for example, https://github.com/nebari-dev/nebari/blob/215bbd622f0ea1692e79bb4411db81ef28b1d329/src/_nebari/upgrade.py#L1032-L1036


The idea of having a `--force` flag is to clearly define what the default behavior should be, without any user input, to upgrade Nebari. That will allow us to run automatic upgrade tests in our CI/CD workflows and improve our confidence in the changes we incorporate.

There's currently a [PR ready to be reviewed](https://github.com/nebari-dev/nebari/pull/2780) implementing the logic to add these automatic upgrade tests to our GHA workflows. You'll notice it's actually using the `nebari upgrade` command and passing the `--attempt-fixes` flag: https://github.com/nebari-dev/nebari/blob/09e34a5e3f3fa48062f0ec64a430f2ab973d4fb7/.github/workflows/test_local_upgrade.yaml#L73

This would be the equivalent of having a `--force` flag and is already implemented and fully exposed to the upgrade steps. For example: https://github.com/nebari-dev/nebari/blob/3f2523aae801fa5d09c414aec6fe1eb237effc0f/src/_nebari/upgrade.py#L485-L494

Now, what we need to do, is to make sure that recent upgrade steps that prompt the user for input can be run automatically by defining a default logical path using this flag.


Let me know if you're still interested in contributing a fix for this issue and I'll be happy to answer any questions you might have.


@viniciusdc does it make sense to edit the issue body and title to reflect the fact that we don't need an extra flag as we already have one implemented? It's just a matter of actually using it on the existing and coming upgrade steps.

**Comment by viniciusdc** [comment]
> @viniciusdc does it make sense to edit the issue body and title to reflect the fact that we don't need an extra flag as we already have one implemented? It's just a matter of actually using it on the existing and coming upgrade steps.

Yes, I think it does. The renaming will still be kept in history if needed.

**Comment by marcelovilla** [comment]
Hey @Suraj-kumar00. Just reaching out to see if you're still interested in picking up this issue.
